plugins/videobasedtracker/UVCImageSource.cpp

```diff
@@ -84,13 +84,25 @@ namespace vbtracker {
         //@}
 
       private:
-        uvc_context_t *uvcContext_;
-        uvc_device_t *camera_;
-        uvc_device_handle_t *cameraHandle_;
+        typedef std::unique_ptr<uvc_frame_t, decltype(&uvc_free_frame)>
```

Comment:
So this is a stateful deleter, so it's larger/slower than a single pointer. Something akin to the following would let you skip the second constructor parameter to `unique_ptr` and would remove the state (and thus overhead) from the smart pointer.

```c++
struct UVCFrameDeleter {
void operator()(uvc_frame_t * ptr) { uvc_free_frame(ptr); }
};

using FramePtr = std::unique_ptr<uvc_frame_t, UVCFrameDeleter>;
```